instance/models/openedx_instance.py

```diff
@@ -0,0 +1,197 @@
+class OpenEdXInstance(Instance, OpenEdXAppConfiguration, OpenEdXDatabaseMixin, OpenEdXStorageMixin):
+    """
+    OpenEdXInstance: represents a website or set of affiliated websites powered by the same
+    OpenEdX installation.
+    """
+
+    # Most settings/fields are inherited from OpenEdXAppConfiguration
+    sub_domain = models.CharField(max_length=50)
+    base_domain = models.CharField(max_length=50, blank=True)
+
+    active_appserver = models.OneToOneField(
+        OpenEdXAppServer, null=True, blank=True, on_delete=models.SET_NULL, related_name='+',
+    )
+
+    class Meta:
+        unique_together = ('base_domain', 'sub_domain')
+        verbose_name = 'Open edX Instance'
+
+    @property
+    def domain(self):
+        """
+        Instance domain name
+        """
+        return '{0.sub_domain}.{0.base_domain}'.format(self)
+
+    @property
+    def url(self):
+        """
+        LMS URL
+        """
+        return u'{0.protocol}://{0.domain}/'.format(self)
+
+    @property
+    def studio_sub_domain(self):
+        """
+        Studio sub-domain name (eg. 'studio.master')
+        """
+        return 'studio.{}'.format(self.sub_domain)
+
+    @property
+    def studio_domain(self):
+        """
+        Studio full domain name (eg. 'studio.master.sandbox.opencraft.com')
+        """
+        return '{0.studio_sub_domain}.{0.base_domain}'.format(self)
+
+    @property
+    def studio_url(self):
+        """
+        Studio URL
+        """
+        return u'{0.protocol}://{0.studio_domain}/'.format(self)
+
+    @property
+    def database_name(self):
+        """
+        The database name used for external databases/storages, if any.
+        """
+        name = self.domain.replace('.', '_')
+        # Escape all non-ascii characters and truncate to 64 chars, the maximum for mysql:
+        allowed = string.ascii_letters + string.digits + '_'
+        escaped = ''.join(char for char in name if char in allowed)
+        return truncate_name(escaped, length=64)
+
+    def set_field_defaults(self):
+        """
+        Set default values.
+        """
+        # Main settings
+        from instance.models.openedx_appserver import DEFAULT_EDX_PLATFORM_REPO_URL
+        if not self.openedx_release:
+            self.openedx_release = settings.DEFAULT_OPENEDX_RELEASE
+        if not self.configuration_source_repo_url:
+            self.configuration_source_repo_url = settings.DEFAULT_CONFIGURATION_REPO_URL
+        if not self.configuration_version:
+            self.configuration_version = settings.DEFAULT_CONFIGURATION_VERSION
+        if not self.edx_platform_repository_url:
+            self.edx_platform_repository_url = DEFAULT_EDX_PLATFORM_REPO_URL
+        if not self.edx_platform_commit:
+            self.edx_platform_commit = self.openedx_release
+
+        # Database settings
+        OpenEdXDatabaseMixin.set_field_defaults(self)
+
+        # Storage settings
+        OpenEdXStorageMixin.set_field_defaults(self)
```

Comment:
@bradenmacdonald It's necessary to call them explicitly because the leftmost superclass of `OpenEdXInstance` that defines `set_field_defaults` is `ValidateModelMixin`, and that's what `super().set_field_defaults()` ends up calling.